### DISCUSSION TITLE
Binning problem

### DISCUSSION THREAD

**Discussion by rsneumann** [jtablesaw/tablesaw#1054] [Q&A] [answered]
Hi, given the following table:

Item	Price	Quantity
A	13	2
B	12	12
C	9	45
D	11	32

I want to obtain the mean Price for intervals of Quantity. 
For example, for intervals of 15:
Quantity  1-15: (13+12)/2=12.5
Quantity 16-30: 0
Quantity 31-45: (9+11)/2=10
How can I achieve this?
(The userguide has an example using binning that does not compile)

**Comment by lwhite1** [comment, marked as answer]
The first thing that comes to mind is to create a new column for the quantity intervals and then use mapInto() which takes as arguments a function and and the new column. It's something like 

quantity.mapInto(myFunctionThatTakesAnIntAndReturnsAString, myOutputColumn)

Then you can just use the usual summarize().by() to do the calculations.

**Reply by rsneumann** [reply]
Hi - I got it to work as you suggested - thank you!
